### ISSUE TITLE
Questions about `OneRestore trained on our CDD-11 for Real Scenes.`

### ISSUE THREAD

**Issue by jkhu29** [gy65896/OneRestore#13]
In the README, you mentioned a pre-trained model named `OneRestore trained on our CDD-11 for Real Scenes.` However, your paper didn't provide details of the training of this model. What is `for Real Scenes`? Does it require a special dataset or a new loss function? How is it different from the training settings described in Sec.5.1 of the paper?

**Comment by YuanGao-YG** [comment]
Thanks for your attention to our work. The pre-trained model named OneRestore trained on our CDD-11 for Real Scenes (onerestore_real.tar) was also trained using CDD-11. It doesn't require a special dataset or a new loss function. It follows the same training settings described in Sec.5.1 of the paper.

**Comment by jkhu29** [comment]
Thanks for your reply, but I have a further question. What is the difference between `[onerestore_cdd-11.tar](https://1drv.ms/u/s!As3rCDROnrbLgqpmWkGBku6oj33efg?e=7yUGfN)` and `[onerestore_real.tar](https://1drv.ms/u/s!As3rCDROnrbLgqpi-iJOyN6OSYqiaA?e=QFfMeL)`. It seems that they are both trained on CDD11, with the same loss function, epochs, learning-rate, optimizer, etc.

**Comment by bozoct** [comment]
Same question here.

**Comment by gy65896** [comment]
Thank you for your question. Specifically, both models are trained under the same configuration. The difference is that for the first model, we selected the result with the best indicator for experimental comparison, while the other one is the model of the last batch.

**Comment by jkhu29** [comment]
So, `onerestore_cdd-11.tar` is the best parameter for CDD11-testset while `onerestore_real.tar` is the parameter of the last epoch. But why the parameter in the last epoch can be **for Real Scenes**?

**Comment by gy65896** [comment]
In practice, this is not different, but since we used different model weights for synthetic and real data in the paper, we publish both weights to ensure that the implementation results match our paper.

**Comment by jkhu29** [comment]
Could you explain why you choose to use the parameters of the final epoch for real data?

**Comment by gy65896** [comment]
There is no special reason, we publish this just to allow researchers to reproduce the same results in our paper.

**Comment by jkhu29** [comment]
Well, let me describe my query in Chinese...

根据您的解释，OneRestore并没有使用额外的训练方式、新增的退化或数据集，仅仅使用最后一轮的的权重去处理真实退化场景的图像。

那么，`onerestore_cdd-11.tar`不能处理真实退化图像吗？如果可以，为什么文章中要采用`onerestore_real.tar`去处理真实图像？如果不行，为什么`onerestore_cdd-11.tar`不能处理真实退化图像？

**Comment by bozoct** [comment]
I guess the authors just did this in implementation but not with a clear purpose. It may be okay to use the same weights for real scenes. Have you tested the onerestore_cdd-11.tar for real scenes?

> Well, let me describe my query in Chinese...
> 
> 根据您的解释，OneRestore并没有使用额外的训练方式、新增的退化或数据集，仅仅使用最后一轮的的权重去处理真实退化场景的图像。
> 
> 那么，`onerestore_cdd-11.tar`不能处理真实退化图像吗？如果可以，为什么文章中要采用`onerestore_real.tar`去处理真实图像？如果不行，为什么`onerestore_cdd-11.tar`不能处理真实退化图像？

**Comment by gy65896** [comment]
You are right. As I mentioned before, these two checkpoints have no significant difference in model training. It does not mean that onerestore_cdd-11.tar can not process the real image. We just want researchers can produce the same results as our paper. Please don't dwell on this.

**Comment by jkhu29** [comment]
> I guess the authors just did this in implementation but not with a clear purpose. It may be okay to use the same weights for real scenes. Have you tested the onerestore_cdd-11.tar for real scenes?
> 
> > Well, let me describe my query in Chinese...
> > 根据您的解释，OneRestore并没有使用额外的训练方式、新增的退化或数据集，仅仅使用最后一轮的的权重去处理真实退化场景的图像。
> > 那么，`onerestore_cdd-11.tar`不能处理真实退化图像吗？如果可以，为什么文章中要采用`onerestore_real.tar`去处理真实图像？如果不行，为什么`onerestore_cdd-11.tar`不能处理真实退化图像？

This implementation confuses me a lot. The authors just do this but don't clearly state the reason for using this setting in their paper or repository description, which makes me doubt these results of this paper.

We may never know the authors' intention in conducting the test for real data in such way.

**Comment by gy65896** [comment]
If you have any doubts, you can just do the experiment. I have made all the checkpoints public.

**Comment by gy65896** [comment]
Please do not give unreasonable responsibilities.

**Comment by jkhu29** [comment]
Dear authors of OneRestore,

**It is the author's responsibility to fully explain the setting of your paper and the intention behind it.**

And your "strongly worded" feedback will cause more researchers to misunderstand your work. You keep ducking this issue for no reasons.

**Comment by jkhu29** [comment]
Maybe you just conduct the wrong test, or maybe there is a scientific research topic behind this: perhaps extended training time, appropriate parameter selection or stacking can increase the generalization of image restoration models under complicated and heterogeneous real degradations? 

As readers, we cannot know why, and it is difficult to fully replicate your experiments. We have our own experiments. We desire an explanation, even if it is due to your errors or omissions. Explanation is preferable to evading the situation.

Perhaps you could wait until you are emotionally stable before answering the questions above.

**Comment by bozoct** [comment]
I just compared two weights on real scenes and obtained very similar results.

**Comment by jkhu29** [comment]
So they just ran experiments in the wrong settings and refused to explain why they did so. Very interesting. What a fantastic work! Using purely simulated data can solve real degradation problems, which is very worthy of study.

**Comment by jkhu29** [comment]
This is the first time I've seen various weights used to test different tasks without explanation.

Perhaps this involves an advanced form of scientific writing.

**Comment by bozoct** [comment]
Regarding this, I am also confused of the good performance on real data. I ran my algorithm on the CDD-11 dataset. The performance on the test set is OK but generalization to real data is poor. I doubt if it is due to the explicit prompt that benefits the generalization.

> So they just ran experiments in the wrong settings and refused to explain why they did so. Very interesting. What a fantastic work! Using purely simulated data can solve real degradation problems, which is very worthy of study.

**Comment by jkhu29** [comment]
> Regarding this, I am also confused of the good performance on real data. I ran my algorithm on the CDD-11 dataset. The performance on the test set is OK but generalization to real data is poor. I doubt if it is due to the explicit prompt that benefits the generalization.
> 
> > So they just ran experiments in the wrong settings and refused to explain why they did so. Very interesting. What a fantastic work! Using purely simulated data can solve real degradation problems, which is very worthy of study.

Well, I guess they just picked the best test set for their model, which is exactly what I doubt in this paper.

BTW, I think these two authors will not answer this question seriously, they will say "just do the experiments".

**Comment by jkhu29** [comment]
To be honest, these authors are quite cute, like children who do something wrong but try hard to duck the issues and make excuses.

**Comment by jkhu29** [comment]
> Regarding this, I am also confused of the good performance on real data. I ran my algorithm on the CDD-11 dataset. The performance on the test set is OK but generalization to real data is poor. I doubt if it is due to the explicit prompt that benefits the generalization.
> 
> > So they just ran experiments in the wrong settings and refused to explain why they did so. Very interesting. What a fantastic work! Using purely simulated data can solve real degradation problems, which is very worthy of study.

[InsturctIR](https://github.com/mv-lab/InstructIR) also indicates that with language prompts, the performance of the restoration model does not appear to decline as the number of degradation types in the all-in-one task increases. It may be the case that including text information helps the restoration model to generalize.

**Comment by XiaoqiHu-zju** [comment]
I am a researcher in this field, and I believe the author did nothing wrong. Using results from training on simulated data to handle real situations is reasonable; this is a common practice in low-level vision tasks. Otherwise, do you have a better method to directly collect real paired data for various degradation situations? If you do, you will be the greatest contributor in this field. By the way, after reading the entire review, it seems that the one losing control of their emotions is you. You have been constantly despising and insulting the author, while the author just chose to ignore you. No offense, but I hope you don't encounter a reviewer like yourself when you submit your paper.

**Comment by jkhu29** [comment]
Since RealESRGAN, two-order simulated data is used to help the restoration model to generalize on real degradations. Everyone knows it. GAN or generation prior (like diffusion model) also helps. Could you (@XiaoqiHu-zju ) show me some articles proving that the restoration model can generalize well to real data from only single-order degraded image without GAN or diffusion prior?

I don't think I would like to review a paper with wrong experimental settings. Even if I review it, I will point out the illogical places to help it be more perfect.

Or, do you think it is reasonable to use different weights to test different tasks without explanation? If so, I have nothing to say.

In addition, maybe you don't know, **this author deleted some very offensive comments.** I am also curious about why he deleted them. Hi, @gy65896 , could you tell me why you delete those comments?

> I am a researcher in this field, and I believe the author did nothing wrong. Using results from training on simulated data to handle real situations is reasonable; this is a common practice in low-level vision tasks. Otherwise, do you have a better method to directly collect real paired data for various degradation situations? If you do, you will be the greatest contributor in this field. By the way, after reading the entire review, it seems that the one losing control of their emotions is you. You have been constantly despising and insulting the author, while the author just chose to ignore you. No offense, but I hope you don't encounter a reviewer like yourself when you submit your paper.

**Comment by jkhu29** [comment]
Hi, @XiaoqiHu-zju, could you proving some papers about the restoration model can generalize well to real data from only single-order degraded image without GAN or diffusion prior? If not, I will explain the whole story to you.

At the beginning, I question the settings of this paper, that is, different parameters should not be chosen for different tasks without explaining the reasons; even if this is done, the author should add relevant explaination and attach them to the appendix.

Then I was shocked by the author's attitude. They refused to explain, saying "just do the experiments", and even said in the offensive comments he deleted that they were not obliged to explain this issue further. ***Am I obliged to take responsibility for the soundness of your paper?***

Based on the author's strange reaction, I began to question the authenticity of the real data experiment in this article. When zzzbzzz did some experiments and proved that it could be just a simple wrong experimental setting, I withdrew my accusation against the conclusion of the article and turned my doubt to the selection of data, as I said [here](https://github.com/gy65896/OneRestore/issues/13#issuecomment-2513737290). Strictly speaking, these are some unavoidable minor problems in scientific research.

Then, @XiaoqiHu-zju, you appeared and expressed some opinions. I fully respect your right to express your opinions, and I ask you to make comments after understanding the context of the matter.

**Comment by jkhu29** [comment]
> I am a researcher in this field, and I believe the author did nothing wrong. Using results from training on simulated data to handle real situations is reasonable; this is a common practice in low-level vision tasks. Otherwise, do you have a better method to directly collect real paired data for various degradation situations? If you do, you will be the greatest contributor in this field. By the way, after reading the entire review, it seems that the one losing control of their emotions is you. You have been constantly despising and insulting the author, while the author just chose to ignore you. No offense, but I hope you don't encounter a reviewer like yourself when you submit your paper.

@XiaoqiHu-zju Another borderline reject reviewer who does not engage in any discussion with the one of detailed responses. No offense, but I hope I don't encounter a reviewer like you.

**Comment by gy65896** [comment]
> In addition, maybe you don't know, **this author deleted some very offensive comments.** I am also curious about why he deleted them. Hi, @gy65896 , could you tell me why you delete those comments?

Dear commenter, in theory, every reply comment will be submitted to your email address. I remember that no comment was deleted. You can check it. I don't know why you are so angry.

**Comment by gy65896** [comment]
In theory, all my models have been submitted, and you can indeed get the same results as in the paper by running my model. It is true that you said that we used two models to process real and synthetic scenes respectively, but why is this wrong? I don’t quite understand. I feel that there is no problem with my experiment. Thank you for your comment. I am sorry if that sentence makes you unhappy.

**Comment by gy65896** [comment]
I'm sorry that I didn't reply to your message because I had an important speech to give. It's my problem, but I also ask you to understand that not everyone needs to revolve around you.

**Comment by jkhu29** [comment]
> > In addition, maybe you don't know, **this author deleted some very offensive comments.** I am also curious about why he deleted them. Hi, @gy65896 , could you tell me why you delete those comments?
> 
> Dear commenter, in theory, every reply comment will be submitted to your email address. I remember that no comment was deleted. You can check it. I don't know why you are so angry.

I checked my e-mail and found that not every reply is recorded. It's strange. For example, this reply did not appear in my mailbox, but I can see it in github.

**Comment by gy65896** [comment]
I should have edited my reply right away because I think that was inappropriate. Did you say it took me a long time to delete it?

**Comment by jkhu29** [comment]
> In theory, all my models have been submitted, and you can indeed get the same results as in the paper by running my model. It is true that you said that we used two models to process real and synthetic scenes respectively, but why is this wrong? I don’t quite understand. I feel that there is no problem with my experiment. Thank you for your comment. I am sorry if that sentence makes you unhappy.

I think I have been allowed to ask you more clearly and directly. 

*I don’t think it’s wrong to use two models for different tasks.* I think it’s wrong to use two models for different tasks `without an explanation`. I did not find any explanation in the article.

I am curious about the generalization of the restored model to real degradations. If complex degradations can help the generalization in some context, I think it's a really huge contribution.

**Comment by gy65896** [comment]
In my experiments, the generalization ability for low light and fog is still good, but for rain and snow scenes, our generalization for real scenes is limited.

**Comment by jkhu29** [comment]
> I should have edited my reply right away because I think that was inappropriate. Did you say it took me a long time to delete it?

No. I have the impression that you posted a comment in a very strong tone accusing me of verifying the above issues as my own responsibility. I can't understand why you posted this at that time.

**Comment by gy65896** [comment]
In fact, this is a misunderstanding. When we used the model, my collaborator and I tested the simulation and real scenes respectively. We just used the different model. There is no deeper reason.
I also agree with your point of view that better generalization of real scenarios is the ultimate ideal of this type of work, and I am also committed to solving it.

**Comment by gy65896** [comment]
Can you take a screenshot? I don’t think I saw it, or it’s the one I edited immediately?

**Comment by jkhu29** [comment]
> In fact, this is a misunderstanding. When we used the model, my collaborator and I tested the simulation and real scenes respectively. We just used the different model. There is no deeper reason. I also agree with your point of view that better generalization of real scenarios is the ultimate ideal of this type of work, and I am also committed to solving it.

Got it. After my own test and zzzbzzz's test, I have realized this [here](https://github.com/gy65896/OneRestore/issues/13#issuecomment-2513718381).

**Comment by gy65896** [comment]
I remembered that you gave me a bad emoj in my reply. I was angry and wrote the first one, but I changed it immediately.

**Comment by jkhu29** [comment]
> > I should have edited my reply right away because I think that was inappropriate. Did you say it took me a long time to delete it?
> 
> No. I have the impression that you posted a comment in a very strong tone accusing me of verifying the above issues as my own responsibility. I can't understand why you posted this at that time.

But, if it's really deleted, I can't see it anymore. After this [comment](https://github.com/gy65896/OneRestore/issues/13#issuecomment-2510867049).

**Comment by jkhu29** [comment]
> I remembered that you gave me a bad emoj in my reply. I was angry and wrote the first one, but I changed it immediately.

I think that is the reason.

**Comment by gy65896** [comment]
It's okay, since it's just a misunderstanding, just talk it out. Thank you for your attention to my work.

**Comment by gy65896** [comment]
I hope I solved your doubts.

**Comment by jkhu29** [comment]
Great work finally. My issue was solved in a rather tortuous path. Nighty night.

I also apologize for my previous inappropriate remarks.

**Comment by gy65896** [comment]
It doesn't matter, life is interesting because of all the weird misunderstandings.
